src/moonlink_connectors/src/rest_ingest/decimal_utils.rs

```diff
@@ -0,0 +1,190 @@
+use bigdecimal::BigDecimal;
+use moonlink::row::RowValue;
+use std::convert::TryInto;
+use std::str::FromStr;
+use thiserror::Error;
+
+#[derive(Debug, Error, PartialEq)]
+pub enum DecimalConversionError {
+    #[error("Decimal normalization precision failed (value: {value}, parsed precision: {parsed_precision})")]
+    NormalizationPrecision {
+        value: String,
+        parsed_precision: usize,
+    },
+    #[error("Decimal normalization scale failed (value: {value}, parsed scale: {parsed_scale})")]
+    NormalizationScale { value: String, parsed_scale: i64 },
+    #[error("Decimal precision exceeds the specified precision (value: {value}, expected ≤ {expected_precision}, actual {actual_precision})")]
+    PrecisionMismatch {
```

Comment:
I won't call it "mismatch", because precision is supposed to represent **max** number of digits, instead of a perfect match.